include/fiction/networks/views/extended_rank_view.hpp

```diff
@@ -398,6 +410,33 @@ class extended_rank_view<Ntk, false> : public fiction::depth_view<Ntk>
                   [this](auto const& n1, auto const& n2) { return rank_pos.at(n1) < rank_pos.at(n2); });
         mockturtle::detail::foreach_element(pis.cbegin(), pis.cend(), std::forward<Fn>(fn));
     }
+
+    /**
+     * Applies a given function to each PI in unranked order, so the order in the underlying `depth_view`.
+     *
+     * @tparam Fn Functor type.
+     * @param fn The function to apply.
+     */
+    template <typename Fn>
+    void foreach_pi_unranked(Fn&& fn) const
+    {
+        fiction::depth_view<Ntk>::foreach_pi(std::forward<Fn>(fn));
+    }
+
+    /**
+     * Rearranges the rank order of the PIs to the underlying PI order.
+     * This can be used to ensure equivalent PI order for equivalence checking.
+     */
+    void rearrange_pis()
+    {
+        std::vector<node> pis{};
+        pis.reserve(this->num_pis());
+
+        fiction::depth_view<Ntk>::foreach_pi([&pis](auto const& pi) { pis.push_back(pi); });
+
+        set_ranks(0, pis);
+    }
```

Comment:
Similarly here. I understand this function rearranges the PI order somehow. Are the PIs ordered by their order of creation? This isn't quite clear to me yet.